drivers/dma/dma_mcux_edma.c

```diff
@@ -275,6 +274,27 @@ static void dma_mcux_edma_error_irq_handler(const struct device *dev)
 	barrier_dsync_fence_full();
 #endif
 }
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(channels_shared_irq_mask)
+static void dma_mcux_edma_multi_channels_irq_handler(const struct device *dev, uint32_t idx,
+			uint32_t *buf, uint32_t step)
+{
+	uint32_t index = (step * idx);
+	uint32_t num = buf[index];
+	uint32_t count = 0;
+
+	for (int _i = 0; _i < step; _i++) {
+		while (num > 0) {
+			if ((num & 0x1) == 1) {
+				dma_mcux_edma_irq_handler(dev, count);
+			}
+			num = num >> 1;
+			count++;
+		}
+		index += _i;
```

Comment:
Why add to index like this? Wouldn't we simply want to increment index by 1 each iteration? We could also remove index entirely and simply move the `buf` pointer.

Reply:
Updated. Could you help check again?